addons/account/views/account_payment_view.xml

```diff
@@ -137,8 +137,11 @@
                                     attrs="{'invisible':[('move_line_ids','=',[])]}" icon="fa-bars"/>
                             <field name="move_line_ids" invisible="1"/>
                             <button class="oe_stat_button" name="button_invoices"
-                                    string="Invoices" type="object"
-                                    attrs="{'invisible':[('has_invoices','=',False)]}" icon="fa-bars"/>
+                                    string="1 Invoice" type="object"
```

Comment:
Seriously? :p So, if the payment is reconciled with more than one invoice, we display 1? This seems plain wrong.

Why not taking inspiration from this kind of stuff ? https://github.com/odoo/enterprise/blob/12.0/account_sepa_direct_debit/views/sdd_mandate_views.xml#L21

You will then just need to count the number of invoices reconciled with your payment. You could add a computed field for that, and simply use _compute_reconciled_invoice_ids of account.payment to compute it.